# Tarea 1 - MEL

Alejandro Mantilla - 201711304

Ximena Palacio - 201730995

Juan Manuel Betancourt - 201632544

## Índice

- [Parte A](#Parte-A.-Problemas-Teóricos-y-Conceptuales.)
    - [Problema 1](#Problema-1.-Fundamentos-de-Inferencia-Paramétrica.)
        - [__I.__](#I.-$\bar{Y}-=-\frac{1}{n}\sum_{i=1}^{n}Y_{i}-\sim-\text{Normal}(0,1/n)$)
        - [__II.__](#II.-$\sum_{i=1}^{n}Y_{i}^{2}-\sim-\chi^{2}(n)$)
        - [__III.__](#III.-$\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2}-\sim-\chi^{2}(n-1)$)
        - [__IV.__](#IV.-$c-\cdot-\frac{n\bar{Y}^{2}}{\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2}}-\sim-F(1,-n-1)$;-con-$c-=-?$)
    - [Problema 2](#Problema-2.-Fundamentos-Distribuciones-Multivariadas:-Propiedades.)
    - [Problema 3](#Problema-3.-Fundamentos-de-Inferencia-Paramétrica.)
    - [Problema 4](#Problema-4.-Fundamentos-de-Inferencia-Paramétrica.)
        - [__I.__](#I.-Muestre-que-$\sum_{i=1}^{n}a_{i}-=-1$.)
        - [__II.__](#II.-Demuestre-que-de-todos-los-posibles-estimadores-insesgados-lineales-para-$\mu$,-$\bar{Y}$-es-el-que-tiene-menor-varianza-(BLUE).)
    - [Problema 5](#Problema-5.-Fundamentos-del-Modelo-de-Regresión-Lineal.)
        - [__I.__](#I.-Sí-$p=n$,-qué-relación-hay-entre-$Y$-y-$\hat-Y$?)
        - [__II.__](#II.-Porque-($Y$-y-$\hat-Y$)-debe-ser-ortogonal-a-cualquier-$\omega-\in-\Omega$)
        - [__III.__](#III.-Ahora-suponga-que-se-hace-una-transformación-lineal-para-crean-nuevas-variables-${Z}$-tal-que-la-nueva-matriz-de-datos-es:-$Z=XA$,-donde-$A-\in-\mathbb{R}^{p\times-p}$-es-una-matriz-de-rango-completo.-Sí-se-define--$\Omega_z-\{Z\theta-\vert{\theta}\in-\mathbb{R}^p-\}$-entonces,-¿qué-relación-existe-entre\-$\hat-Y$-y-la-proyección-de-$Y$-en-$\Omega_z$?)
        - [__IV.__](#IV.-Con-el-resultado-anterior,-y-usando-los-datos-measure.txt,-se-usa-el-contorno-del-pecho-como-la-variable-de-respuesta-$Y$-y-las-otras-dos-como-variables-independientes-$X_1$-y-$X_2$.-Se-estima-el-modelo-de-regresión-correspondiente.)
            - [__a.__](#a.-----Se-quiere-encontrar-los-vectores-$\hat{Y}$-con-los-modelos-que-tiene-las-variables-$X$-y-las-variables-$W$)
            - [__b.__](#b.-Contrastar-un-intervalo-de-confianza-para-el-valor-esperado-del-contorno-del-pecho)
    - [Problema 6](#Problema-6.-Fundamentos-de-Álgebra-para-Modelos-Lineales.)
        - [__I.__](#I.-$\frac{d\alpha_1}{dz}-=-a$)
        - [__II.__](#II.-$\frac{d\alpha_2}{dz}-=-(\mathbb{M}-+-\mathbb{M}^T)z$.)
    - [Problema 7](#Problema-7.-Fundamentos-de-Inferencia-Paramétrica.)
        - [__I.__](#I.-Halle-el-estimador-de-máxima-verosimilitud-$\hat{\theta}$)
        - [__II.__](#II.-----Comente-sobre-la-distribución-de-$\frac{\hat{\theta}-{\theta}}{\sqrt{\left(\frac{{\theta}^\mathbf{2}}{t{n}}\right)}\-}\-$cuando-$n$-representa-un-tamaño-de-muestra-grande.)
        - [__III.__](#III.-Consulte-sobre-el-teorema-de-Slutsky-y-úselo-junto-con-la-información-del-numeral-anterior-para-construir-un-intervalo-de-confianza-$(1-\alpha)$-para-el-parámetro-$\theta$.)
        - [__IV.__](#IV.-----Suponiendo-que-el-número-de-entradas-que-ocurren-al-baño-del-5-piso-del-edificio-ML,-se-comportan-como-un-proceso-Poisson-con-tasa-$\frac{\mathbf{1}}{\mathbf{\theta}}$.-Y-suponiendo-que,-en-una-semana,-el-promedio-de-entradas-por-hora-es-de-21.4-y-que-el-día-laboral-tiene-12-horas.--Se-construye-un-intervalo-de-confianza-para-la-media-del-tiempo-entre-arribos-con-una-confianza-del-95%)
    - [Problema 8](#Problema-8.-Proyecciones-y-Distribuciones.)
        - [__I.__](#)
        - [__II.__](#)
        - [__III.__](#)
- [Parte B](#Parte-B.-Problemas-Aplicados-con-Datos-Reales.)
    - [Problema 9](#Problema-9.-Carseats-Sales-Data.)
        - [__I.__](#)
        - [__II.__](#)
        - [__III.__](#)
        - [__IV.__](#)
        - [__V.__](#)
        - [__VI.__](#)
        - [__VII.__](#)
        - [__VIII.__](#)
    - [Problema 10](#Problema-10.-Problema-Libre.)
        - [__I.__](#I.----Haga-una-descripción-de-los-datos-y-del-problema-general-sobre-el-cual-quiere-inferir.-Muestre-algunas-gráficas-que-resuman-el-comportamiento-de-los-datos.)
        - [__II.__](#II.----Se-realizan-las-siguientes-pruebas)
            - [__a.__](#a.-Significancia-individual-de-cada-variable)
            - [__b.__](#b.-Significancia-global-del-modelo)
            - [__c.__](#c.-Significancia-parcial-de-algunas-variables)
            - [__d.__](#d.-Una-combinación-lineal-de-betas-(que-tenga-sentido-dentro-del-contexto-del-problema))

## Parte A. Problemas Teóricos y Conceptuales.

### Problema 1. Fundamentos de Inferencia Paramétrica.

Ayuda:
* Si $Z \sim \text{Normal}(0, 1)$, entonces $Z^{2}\sim \chi^{2}(1)$.
* Si $Z_{1} \sim \chi^{2}(v_{1})$ y $Z_{2} \sim \chi^{2}(v_{2})$ y  $Z_{1}$ y $Z_{2}$ son independientes, entonces $(Z_{1} + Z_{2}) \sim\chi^{2}(v_{1} + v_{2})$.
* $\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2}$ y $\bar{Y}$ son independientes en el caso de $Y_{i} \sim \text{Normal}(\mu,\sigma^{2})$.

Se tiene una muestra aleatoria $Y_1, Y_2, \cdots , Y_n$ de una población $Y \sim \text{Normal}(0, 1)$. Demuestre:

#### __I.__ $\bar{Y} = \frac{1}{n}\sum_{i=1}^{n}Y_{i} \sim \text{Normal}(0,1/n)$

Partimos de que

$$
Y_{i}\sim \text{Normal}(0,1).
$$

Por propiedad de la varianza de la distribución normal, multiplicar la variable por un factor, afecta la distribución de manera que la varianza se multiplica por el cuadrado del factor.

\begin{align*}
    \frac{Y_{i}}{n} &\sim \text{Normal}(0,1/{n^{2}})\\
    \frac{Y_{1}}{n} + \frac{Y_{2}}{n} + \cdots + \frac{Y_{n}}{n} &\sim \text{Normal}(0,\underbrace{1/n^{2} + 1/n^{2} + \cdots + 1/n^{2}}_{n \text{ veces}} )\\
    \sum_{i=1}^{n}\frac{Y_{i}}{n} = \frac{1}{n}\sum_{i=1}^{n}Y_{i} &\sim \text{Normal}(0,\sum_{i=1}^{n}1/{n^{2}})\\
    \frac{1}{n}\sum_{i=1}^{n}Y_{i} &\sim \text{Normal}(0,\frac{1}{n}\underbrace{\sum_{i=1}^{n}1/{n}}_{=1})\\
    \frac{1}{n}\sum_{i=1}^{n}Y_{i} &\sim \text{Normal}(0,1/n).
\end{align*}

#### __II.__ $\sum_{i=1}^{n}Y_{i}^{2} \sim \chi^{2}(n)$

Demostraremos por inducción, dadas la primera y segunda ayuda.
        
__Paso base (__$n = 1$__):__

$$
    \sum_{i = 1}^{1}Y_{i}^{2} = Y_{1}^{2} \sim \chi^{2}(1)
$$

La primera ayuda garantiza la afirmación.        

__Paso inductivo (suponer para__ $n = k$__, demostrar para__ $n = k+1$__):__
        
Suponemos que

$$
\sum_{i=1}^{k}Y_{i}^{2} = Y_{1}^{2} + Y_{2}^{2} + \cdots + Y_{k-1}^{2} + Y_{k}^{2} \sim \chi^{2}(k).
$$

Ahora debemos demostrar que

$$
\sum_{i=1}^{k+1}Y_{i}^{2} \sim \chi^{2}(k+1).
$$

Como implicación del supuesto y del paso base, obtenemos que

$$
\sum_{i=1}^{k+1}Y_{i}^{2} = \underbrace{\left[Y_{1}^{2} + Y_{2}^{2} + \cdots + Y_{k-1}^{2} + Y_{k}^{2}\right]}_{\sim \chi^{2}(k)} + \underbrace{\left[Y_{k+1}^{2}\right]}_{\sim \chi^{2}(1)}
$$

Como implicación de la segunda ayuda, obtenemos que

$$
\sum_{i=1}^{k+1}Y_{i}^{2} = \underbrace{\left[Y_{1}^{2} + Y_{2}^{2} + \cdots + Y_{k-1}^{2} + Y_{k}^{2}\right]}_{\sim \chi^{2}(k)} + \underbrace{\left[Y_{k+1}^{2}\right]}_{\sim \chi^{2}(1)} \sim \chi^{2}((k) + (1)) = \chi^{2}(k+1).
$$
        
Con esto, hemos demostrado la afirmación inicial.

 #### __III.__ $\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2} \sim \chi^{2}(n-1)$

Por propiedad de la distribución normal estándar y la primera ayuda, $\bar{Y} \sim \text{Normal}\left(0, \frac{1}{n}\right) \Rightarrow \frac{\bar{Y}}{1/\sqrt{n}} \sim \text{Normal}(0,1) \Rightarrow \left(\frac{\bar{Y}}{1/\sqrt{n}}\right)^{2} \sim \chi^{2}(1)$.

Desarrollemos lo demostrado en el punto anterior, $\sum_{i=1}^{n}Y_{i}^{2} \sim \chi^{2}(n)$.

\begin{align*}
    \sum_{i=1}^{n}Y_{i}^{2} &= \sum_{i=1}^{n}\left(Y_{i} - \bar{Y} + \bar{Y}\right)^{2}\\
    &= \sum_{i=1}^{n}\left(Y_{i} - \bar{Y}\right)^{2} + \sum_{i=1}^{n}\bar{Y}^{2} + 2\bar{Y}\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right).
\end{align*}        

Por propiedad de la muestra de la media, podemos ver que $\sum_{i=1}^{n}Y_{i} = n\bar{Y}$, por lo cual $\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right) = n\bar{Y}-n\bar{Y} = 0$.
        
Seguimos con el desarrollo:

\begin{align*}
    \sum_{i=1}^{n}Y_{i}^{2} &= \sum_{i=1}^{n}\left(Y_{i} - \bar{Y}\right)^{2} + \sum_{i=1}^{n}\bar{Y}^{2} + \underbrace{2\bar{Y}\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)}_{=0}\\
    &= \sum_{i=1}^{n}\left(Y_{i} - \bar{Y}\right)^{2} + \sum_{i=1}^{n}\bar{Y}^{2}\\
    &= \sum_{i=1}^{n}\left(Y_{i} - \bar{Y}\right)^{2} + n\bar{Y}^{2}\\
    &= \sum_{i=1}^{n}\left(Y_{i} - \bar{Y}\right)^{2} + \underbrace{\left(\frac{\bar{Y}}{1/\sqrt{n}}\right)^{2}}_{\sim \chi^{2}(1)}.
\end{align*}        

Ahora despejamos para la suma original cuya distribución queremos conocer y aplicamos la indicación de la segunda ayuda:

\begin{align*}
    \underbrace{\underbrace{\sum_{i=1}^{n}Y_{i}^{2}}_{\sim \chi^{2}(n)} - \underbrace{\left(\frac{\bar{Y}}{1/\sqrt{n}}\right)^{2}}_{\sim \chi^{2}(1)}}_{\sim \chi^{2}(n-1)} &= \sum_{i=1}^{n}\left(Y_{i} - \bar{Y}\right)^{2}.
\end{align*}

#### __IV.__ $c \cdot \frac{n\bar{Y}^{2}}{\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2}} \sim F(1, n-1)$; con $c = ?$

Partimos de la estructura de las variables aleatorias cuya distribución es $F(j, k)$.
        
Sean $W_{1}$ y $W_{2}$ variables aleatorias con distribución $\chi^{2}$ con grados de libertad $j$ y $k$ respectivamente. Entonces,

$$
    \frac{W_{1} / j}{W_{2} / k} \sim F(j, k).
$$  

Sabemos de demostraciones anteriores que

\begin{align*}
    n\bar{Y}^{2} = \left(\frac{\bar{Y}}{1/\sqrt{n}}\right)^{2} &\sim \chi^{2}(1)\text{, y que}\\
    \sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2} &\sim \chi^{2}(n-1).
\end{align*}

Haciendo uso de la tercera ayuda, debido a que el numerador y el denominador son independientes, podemos afirmar la siguiente relación:

$$
    \frac{n\bar{Y}^{2} / 1}{\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2} / (n-1)} \sim F(1, n-1).
$$

Concluimos que

$$
    (n-1)\cdot \frac{n\bar{Y}^{2}}{\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2}} \sim F(1, n-1)
$$

y por lo tanto $c=n-1$.

### Problema 2. Fundamentos Distribuciones Multivariadas: Propiedades.

Queremos demostrar que la matriz de covarianzas $Cov(\hat{Y}, Y-\hat{Y})$ contiene ceros únicamente.

Sabemos que la expresión matricial para la estimación de $Y$, es la siguiente:

$$
    \hat{Y} = X\left(X'X\right)^{-1}X'Y = HY.
$$

Podemos expresar $Y-\hat{Y}$ como

\begin{align*}
    Y-\hat{Y} &= Y-HY\\
              &= \left(I - H\right)Y.
\end{align*}

Utilizamos la ayuda del enunciado para expresar la matriz de covarianzas. Note que las matrices $H$ e $I$ son simétricas.

\begin{align*}
    Cov(\hat{Y}, Y-\hat{Y}) &= \,Cov(HY,\left(I - H\right)Y)\\
                            &= H\,Cov(Y,Y)\left(I - H\right)'\\
                            &= H\,Cov(Y,Y)\left(I' - H'\right)\\
                            &= H\,Cov(Y,Y)\left(I - H\right).
\end{align*}

El supuesto de que las observaciones son independientes entre sí nos permite que $Cov(Y,Y)=I$.

\begin{align*}
    Cov(\hat{Y}, Y-\hat{Y}) &= H\,Cov(Y,Y)\left(I - H\right)\\
                            &= H\,\left(I - H\right)\\
                            &= H - H^{2}.
\end{align*}

Por la indempotencia de $H$,

\begin{align*}
    Cov(\hat{Y}, Y-\hat{Y}) &= H-H\\
                            &= 0.
\end{align*}

La matriz de covarianzas contiene solo ceros y por lo tanto $Y$ y $Y-\hat{Y}$ son independientes.

### Problema 3. Fundamentos de Inferencia Paramétrica.

Se tiene una muestra aleatoria $Y_1$, $Y_2$, $\cdots$, $Y_n$ de una población $Y \sim \text{Normal}(\mu, \sigma^{2})$. El estimador convencional de la varianza es:

$$
    S^{2} = \frac{\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2}}{n-1}.
$$

Sin embargo, el estimador por máxima verosimilitud es $\hat{\sigma}^{2}=\frac{(n-1)S^{2}}{n}$. ¿Cuál de los dos es el mejor estimador? 

Iniciamos el procedimiento, desarrollando la expresión del error cuadrático medio. En los siguientes pasos, $E$, se refiere al la función de valor esperado.

\begin{align*}
    MSE\left(\hat{\theta}\right) &= E\left[(\hat{\theta}-\theta)^{2}\right] \\
    &= E\left[(\hat{\theta} - \theta)(\hat{\theta} - \theta)\right] \\
    &= E\left[\hat{\theta}^{2} + \theta^{2} - 2\theta\hat{\theta}\right] \\
    &= E\left[\hat{\theta}^{2}\right] + E\left[\theta^{2}\right] - E\left[2\theta\hat{\theta}\right] \\
    &= E\left[\hat{\theta}^{2}\right] + \theta^{2} - 2\theta E \left[\hat{\theta}\right]
\end{align*}

Expresamos la varianza del estimador en terminos de su valor esperado y despejamos:

\begin{align*}
    & Var\left(\hat{\theta}\right) = E\left[\hat{\theta}^{2}\right] - E\left[\hat{\theta}\right]^{2} \\
    \Rightarrow & E\left[\hat{\theta}^{2}\right] = Var\left(\hat{\theta}\right) + E\left[\hat{\theta}\right]^{2}
\end{align*}

Expresamos el sesgo del estimador en terminos de su valor esperado:

$$
Bias\left(\hat{\theta}\right) = E\left[\hat{\theta}\right] - \theta
$$

Desarrollamos la expresión del $MSE$ según el sesgo y la varianza.

\begin{align*}
    MSE\left(\hat{\theta}\right) &= E\left[\hat{\theta}^{2}\right] + \theta^{2} - 2\theta E \left[\hat{\theta}\right] \\
    &= Var\left(\hat{\theta}\right) + E\left[\hat{\theta}\right]^{2} + \theta^{2} - 2\theta E \left[\hat{\theta}\right] \\
    &= Var\left(\hat{\theta}\right) + \left( E\left[\hat{\theta}\right] - \theta\right)^{2} \\
    &= Var\left(\hat{\theta}\right) + \left( Bias\left(\hat{\theta}\right) \right)^{2}
\end{align*}

A partir de la expresión dada para $S^{2}$ y de manera similar a la demostración del problema [1.3](#3.-$\sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2}-\sim-\chi^{2}(n-1)$):

Sea $X_{i} = Y_{i} - \mu$. Podemos sustituir en $S^{2}$ los $Y_{i}$ por $X_{i}$ ya que restar una constante para cada uno de los $Y_{i}$ no cambia la medida de dispersión.

\begin{align*}
    (n-1)S^{2} &= \sum_{i=1}^{n}\left(Y_{i}-\bar{Y}\right)^{2} = \sum_{i=1}^{n}\left(X_{i}-\bar{X}\right)^{2} \\
    &= \sum_{i=1}^{n}(X_{i}^{2} - 2\bar{X}X_{i} + \bar{X}^{2}) \\
    &= \sum_{i=1}^{n}\left(X_{i}^{2}\right) - 2\bar{X}\sum_{i=1}^{n} \left(X_{i}\right) + n \bar{X}^{2} \\
    &= \sum_{i=1}^{n}X_{i}^{2} - 2n\bar{X}^{2} + n \bar{X}^{2} \\
    &= \sum_{i=1}^{n} X_{i}^{2} - n\bar{X}^{2} \\
    \frac{(n-1)S^{2}}{\sigma^{2}} &= \frac{\sum_{i=1}^{n} X_{i}^{2}}{\sigma^{2}} - \frac{n\bar{X}^{2}}{\sigma^{2}} \\
    &= \sum_{i=1}^{n} \left(\frac{X_{i}}{\sigma}\right)^{2} - n\left(\frac{\bar{X}}{\sigma}\right)^{2} \\
    &= \sum_{i=1}^{n} \left(\frac{X_{i}}{\sigma}\right)^{2} - \left(\frac{\bar{X}}{\sigma/\sqrt{n}}\right)^{2}
\end{align*}

Notamos que $\frac{X_{i}}{\sigma}$ y $\frac{\bar{X}}{\sigma/\sqrt{n}}$ tienen distribución $\text{Normal}(0,1)$ y que el lado derecho de la ecuación puede interpretarse como la suma de $n-1$ variables aleatorias al cuadrado. Como consecuencia de la ayuda 2 del [problema 1](#Problema-1.-Fundamentos-de-Inferencia-Paramétrica.),

$$
    \frac{(n-1)S^{2}}{\sigma^{2}} = \underbrace{\underbrace{\sum_{i=1}^{n} \left(\frac{X_{i}}{\sigma}\right)^{2}}_{\chi^{2}(n)} - \underbrace{\left(\frac{\bar{X}}{\sigma/\sqrt{n}}\right)^{2}}_{\chi^{2}(1)}}_{\chi^{2}(n-1)}.
$$

Por propiedades de la distribución $\chi^{2}(n-1)$, calculamos la varianza de $S^{2}$:

\begin{align*}
    Var\left(\frac{(n-1)S^{2}}{\sigma^{2}}\right) &= 2(n-1) \\
    \frac{(n-1)^{2}}{\sigma^{4}}Var\left(S^{2}\right) &= 2(n-1) \\
    Var\left(S^{2}\right) &= \frac{2\sigma^{4}}{n-1}.
\end{align*}

Calculamos el valor esperado de $S^{2}$:

\begin{align*}
    E\left[\frac{(n-1)S^{2}}{\sigma^{2}}\right] &= n-1 \\
    \frac{n-1}{\sigma^{2}}E\left[S^{2}\right] &= n-1 \\
    E\left[S^{2}\right] &= \sigma^{2}. \\
\end{align*}

Por consecuencia, $Bias\left(S^{2}\right) = 0$ y $MSE\left( S^{2}\right) = \frac{2\sigma^{4}}{n-1}$.

Calculamos la varianza de $\hat{\sigma}^{2}$:

\begin{align*}
    Var\left(\hat{\sigma}^{2}\right) &= Var\left(\frac{(n-1)S^{2}}{n}\right) \\
    &= \left(\frac{n-1}{n}\right)^{2}Var\left(S^{2}\right) \\
    &= \left(\frac{n-1}{n}\right)^{2}\frac{2\sigma^{4}}{n-1} \\
    &= \frac{2\sigma^{4} (n-1)}{n^{2}}. \\
\end{align*}

Calculamos el valor esperado de $\hat{\sigma}^{2}$:

\begin{align*}
    E\left[\hat{\sigma}^{2}\right] &= E\left[\frac{(n-1)S^{2}}{n}\right] \\
    &= \frac{(n-1)}{n}E\left[S^{2}\right] \\
    &= \frac{(n-1)}{n}\sigma^{2}.
\end{align*}

Por consecuencia, $Bias\left(\hat{\sigma}^{2}\right) = \frac{(n-1)}{n}\sigma^{2} - \sigma^{2}$ y 

\begin{align*}
    MSE\left( \hat{\sigma}^{2}\right) &= \frac{2\sigma^{4} (n-1)}{n^{2}} + \left( \frac{(n-1)}{n}\sigma^{2} - \sigma^{2}\right)^{2} \\
    &= \frac{2\sigma^{4} (n-1)}{n^{2}} + \sigma^{4}\left( \frac{(n-1)}{n} - 1\right)^{2} \\
    &= \sigma^{4}\left(\frac{2 (n-1)}{n^{2}} + \left( \frac{(n-1)}{n} - 1\right)^{2}\right) \\
    &= \sigma^{4}\left(\frac{2n - 2}{n^{2}} + \left( \frac{-1}{n}\right)^{2}\right) \\
    &= \sigma^{4}\left(\frac{2n - 1}{n^{2}}\right). \\
    % &= \sigma^{4}\left(\frac{2}{n} - \frac{1}{n^{2}}\right). \\
\end{align*}

Dividimos los $MSE$ para identificar al mayor:

\begin{align*}
    \frac{MSE\left( \hat{\sigma}^{2} \right)}{MSE\left( S^{2} \right)} &= \frac{\frac{2\sigma^{4}}{n-1}}{\sigma^{4}\left(\frac{2n - 1}{n^{2}}\right)} \\
    &= \frac{2\sigma^{4}}{(n-1)\sigma^{4}\left(\frac{2n - 1}{n^{2}}\right)} \\
    &= \frac{2n^{2}}{(n-1)(2n - 1)} \\
    &= \frac{2n^{2}}{2n^{2} -3n + 1}.
\end{align*}

Resulta evidente que $\frac{MSE\left( \hat{\sigma}^{2} \right)}{MSE\left( S^{2} \right)} > 1, \ \forall n \geq 2$, lo que nos permite concluir que $S^{2}$ tiene mejor $MSE$ y, por lo menos para este criterio, es mejor estimador.

Claro está que existen distintos estimadores de la varianza que minimizan el $MSE$, mejor de lo que lo hace $S^{2}$. 

### Problema 4. Fundamentos de Inferencia Paramétrica.

Se tiene una muestra aleatoria $Y_1, Y_2, \cdots, Y_n$ de una población $Y \sim f_{Y}(y)$. Se sabe además que
$\mu = E\left[Y\right]$ y $\sigma^{2} = Var\left(Y\right)$ son dos parámetros desconocidos (media y varianza de la población). Se define un estimador lineal insesgado de $\mu$ como:

$$
    \hat{\mu} = \sum_{i=1}^{n}a_{i}Y_{i},
$$

donde $a_{1}, \cdots,a_{n}$ son constantes arbitrarias y $E[\hat{\mu}] = \mu$.

#### __I.__ Muestre que $\sum_{i=1}^{n}a_{i} = 1$.

Calculamos el valor esperado del estimador $\hat{\mu}$.

\begin{align*}
    E\left[\hat{\mu}\right] &= E\left[ \sum_{i=1}^{n}a_{i}Y_{i} \right] \\
    \mu &= \sum_{i=1}^{n}a_{i} E\left[ Y_{i} \right] \\
    &= \sum_{i=1}^{n}a_{i} \mu \\
    &= \mu \sum_{i=1}^{n}a_{i} \\
    1 &= \sum_{i=1}^{n}a_{i}.
\end{align*}

#### __II.__ Demuestre que de todos los posibles estimadores insesgados lineales para $\mu$, $\bar{Y}$ es el que tiene menor varianza (BLUE).

Calculamos la varianza del estimador $\hat{\mu}$.

\begin{align*}
    Var\left( \hat{\mu} \right) &= E\left[ \hat{\mu}^{2} \right] - E\left[ \hat{\mu} \right]^{2} \\
    &= E\left[ \left(\sum_{i=1}^{n}a_{i}Y_{i}\right)^{2} \right] - E\left[ \sum_{i=1}^{n}a_{i}Y_{i} \right]^{2} \\
    &= E\left[ \sum_{i=1}^{n}\sum_{j=1}^{n}a_{i}a_{j}Y_{i}Y_{j} \right] - \left(\sum_{i=1}^{n}a_{i}E\left[Y_{i}\right]\right)^{2} \\
    &= \sum_{i=1}^{n}\sum_{j=1}^{n}a_{i}a_{j}E\left[ Y_{i}Y_{j} \right] - \sum_{i=1}^{n}\sum_{j=1}^{n}a_{i}a_{j}E\left[Y_{i} \right]E\left[ Y_{j} \right] \\
    &= \sum_{i=1}^{n}\sum_{j=1}^{n}a_{i}a_{j}E\left[ Y_{i}Y_{j} \right] - a_{i}a_{j}E\left[Y_{i} \right]E\left[ Y_{j} \right] \\
    &= \sum_{i=1}^{n}\sum_{j=1}^{n}a_{i}a_{j}\left(E\left[ Y_{i}Y_{j} \right] - E\left[Y_{i} \right]E\left[ Y_{j} \right]\right) \\
    &= \sum_{i=1}^{n}\sum_{j=1}^{n}a_{i}a_{j} Cov(Y_{i}, Y_{j}).
\end{align*}

Las covarianzas entre $Y$s cuando $i\neq j$ son iguales a cero, de manera que se simplifica la expresión:

\begin{align*}
    Var(\hat{\mu}) &= \sum_{i=1}^{n}a_{i}^{2} Var(Y_{i}) \\
    &= \sum_{i=1}^{n} a_{i}^{2} \sigma^{2} \\
    &= \sigma^{2} \sum_{i=1}^{n} a_{i}^{2} \\
\end{align*}

Podemos derivar de manera similar la varianza de $\bar{Y}$:

$$
    Var(\bar{Y}) = \frac{\sigma^{2}}{n}.
$$

Nuestro objetivo es ahora demostrar que $\sum_{i=1}^{n} a_{i}^{2} \geq \frac{1}{n}$. Desarrollemos un poco la desigualdad:

Sean $\vec{a}$ y $\vec{b}$ dos vectores de longitud $n$, separados por un ángulo $\theta$:

$$
    \vec{a} = \left[\begin{matrix} a_{1} & \cdots & a_{n}\end{matrix}\right], \quad \vec{b} = \left[\begin{matrix} 1 & \cdots & 1\end{matrix}\right].
$$

Podemos llegar a la misma expresión desde la desigualdad de Cauchy-Schwarz.

\begin{align*}
    \vec{a}\cdot\vec{b} &\leq \left\Vert\vec{a}\right\Vert \left\Vert\vec{b}\right\Vert \\
    a_{1} + \cdots + a_{n} &\leq \sqrt{\left(a_{1}^{2} + \cdots + a_{n}^{2}\right)} \sqrt{\left(1 + \cdots + 1\right)} \\
    \sum_{i=1}^{n}a_{i}\ &\leq \sqrt{\sum_{i=1}^{n}a_{i}^{2}} \sqrt{n} \\
    \left(\sum_{i=1}^{n}a_{i}\right)^{2} &\leq n\sum_{i=1}^{n}a_{i}^{2} \\
    \frac{1}{n}\left(\sum_{i=1}^{n}a_{i}\right)^{2} &\leq \sum_{i=1}^{n}a_{i}^{2} \\
    \frac{1}{n}\left(1 \right)^{2} &\leq \sum_{i=1}^{n}a_{i}^{2} \\
    \frac{1}{n} &\leq \sum_{i=1}^{n}a_{i}^{2}.
\end{align*}

### Problema 5. Fundamentos del Modelo de Regresión Lineal.

Se tiene un modelo de regresión lineal con $n$ observaciones y $k=p-1$ variables. El estimador de mínimos cuadrados es $\hat\beta \in \mathbb{R} ^p $ y $ \hat Y= X\hat\beta \in \mathbb{R}^n $ es la predicción en los puntos de la muestra. Sea $X=(X_0,X_1,\cdots,X_K ) \in  \mathbb{R} ^{n\times p} $ la matriz de las variables independientes, donde $X_0=(1,1,\cdots,1)$.
Se define entonces el subespacio  $\mathbb{R} ^n$ de todas las posibles combinaciones lineales de $X_0,X_1,\cdots,X_k$ como:

$$ 
    \Omega=\{X\theta \vert \theta \in \mathbb{R} ^p \}.
$$

Por definición, $\Omega$ tiene dimensiones $p \le n.$



#### I. Sí $p=n$, qué relación hay entre $Y$ y $\hat Y$?

Sí se tiene en cuenta que $\hat Y$ es el vector en $\mathbb{R}^n $ que puede ser expresado como una combinación lineal de las columnas de la matriz $X$ que se encuentra más cercano al vector $Y$, lo anterior quiere decir que cuando $p<n$, $\hat Y $ se puede entender como el punto más cercano a  $Y$ sobre $\Omega$. Por lo que el caso en el que $p=n$, la diferencia que hay entre $Y$ y $\hat Y $ se elimina, pues tienen exactamente las mismas dimensiones. Es decir que cuando $p=n$, cualquier vector $Y$ que seleccione ya va a estar sobre el espacio de $\Omega$ y no necesita ser proyectado para poder determinar $\hat Y$, por lo que se puede afirmar que $Y= \hat Y$.

#### II. Porque ($Y$ y $\hat Y$) debe ser ortogonal a cualquier $\omega \in \Omega$

![Imagen punto 5](Archivos/Imagen_Punto_5.png)

La imagen anterior nos permite entender las relaciones entre el plano $\Omega$, $Y$ y $\hat Y$. Se puede ver claramente que $(Y-\hat Y)$ forma un vector entre sí que en la imagen está representado con color rojo punteado. Si se tiene en cuenta que $\hat Y$ es el punto proyectado sobre el plano $\Omega$, más cercano a $Y$, se puede observar que se genera un triángulo rectángulo entre estos dos puntos. Para lo cual, se utiliza el teorema de Pitágoras y con este se puede concluir que el ángulo que hay entre el vector $\hat Y $ y el vector rojo de la resta entre  $(Y-\hat Y)$ debe ser de 90° (ángulo recto) para que la distancia entre estos dos puntos sea la menor, que es lo que se quiere con la proyección.

Teniendo en cuenta lo anterior, es decir que en efecto  $(Y-\hat Y)$ tiene un ángulo recto con respecto a  $\hat Y$, y que  $\hat Y$ es un vector sobre el plano $\Omega$; a cualquier vector sobre este plano se le debe poder replicar lo anterior, por lo tanto, se puede concluir que $(Y-\hat Y)$  es ortogonal a cualquier $\omega \in \Omega$. Esto es coherente con los resultados del problema 2, donde demostramos que ambos vectores tienen covarianza $0$.

#### III. Ahora suponga que se hace una transformación lineal para crean nuevas variables ${Z}$ tal que la nueva matriz de datos es: 

$$
    Z=XA,
$$

donde $A \in \mathbb{R}^{p\times p}$ es una matriz de rango completo.

Si se define 

$$
    \Omega_z \{Z\theta \vert{\theta}\in \mathbb{R}^p \}
$$

entonces, ¿qué relación existe entre $\hat Y$ y la proyección de $Y$ en $\Omega_z$?

Es posible según la definición de $\Omega$ que existan otros vectores que sean combinaciones lineales de las $X$, es decir que por definición del subespacio $\Omega$ es posible escribir efectivamente 
$$
Z=XA
$$

Según lo mencionado en la primera parte de este punto se sabe que $ \hat{Y}$ por definición es una combinación lineal, por lo que en esta transformación lineal para crear Z se sigue cumpliendo lo anteriormente mencionado y $\hat{{Y}}$se puede entender como el punto más cercano a  ${Y}$ sobre $\Omega_z$, pues el subespacio generado por las columnas de $X$ es el mismo que el subespacio generado por las columnas de $W$, dado que $A$ es una matriz que transforma $X$, entonces se cumpliría lo siguiente: 


Sabiendo que $\hat{Y}=X\hat{\beta}=P_\Omega Y$ entonces se hará la comprobación con las matrices de proyección para simplificar la comprobación:
$$
P_X=X\left(X^tX\right)^{-1}X^t
$$

$$
P_X=P_Z
$$


Esto se puede comprobar de la siguiente manera: 
$$
P_z=Z\left(Z^tZ\right)^{-1}Z^t
$$

$$
P_X=X\left(X^tX\right)^{-1}X^t
$$

Y como se sabe que $Z=XA$ …

$$
P_z=XA\left(({XA)}^tXA\right)^{-1}({XA)}^t
$$

Teniendo en cuenta que $A\in{R}^{pxp}$ y $A^t$ y $\ A $ tienen el mismo rango sería una transformación lineal de $\ \mathbb{R}^{nxp} $ a $\ \mathbb{R}^{nxp} $

$$
P_z=XA\left(X^tA^tXA\right)^{-1}({XA)}^t
$$

$$
P_z=XA(A^{-1})\left(X^tX\right)^{-1}\left(A^t\right)^{-1}({XA)}^t
$$

$$
P_z=X\mathbb{I}\left(X^tX\right)^{-1}\mathbb{I}({X)}^t
$$

$$
P_z=X\left(X^tX\right)^{-1}({X)}^t=P_x\ 
$$


Recordando entonces que $\hat{Y}=X\hat{\beta}=P_\Omega Y\$  es claro que: 

$$
\hat{Y}=P_zY=P_xY
$$


#### IV. Con el resultado anterior, y usando los datos measure.txt, se usa el contorno del pecho como la variable de respuesta $Y$ y las otras dos como variables independientes $X_1$ y $X_2$. Se estima el modelo de regresión correspondiente.

Lo primero que se debe hacer es estimar el modelo de regresión correspondiente. Se tiene que el contorno del pecho es la variable $Y$ y las otras variables son las independientes en donde $X_1=\text{cintura} $ y $X_2 =\text{cadera}$ y se sabe que $n=20$.

El modelo de regresión debe ser entonces 
$$
Y=\beta_0+\beta_1x_1+\beta_2x_2+\varepsilon 
$$

$$
Y=X \beta + \varepsilon
$$

In [1]:
#Regresión lineal con X1 y X2
data <- read.table("Archivos/measure.txt")
y <- data$Chest
x1 <- data$Waist
x2 <- data$Hips

RegresionX <- lm(y ~ x1 + x2)
summary(RegresionX)
vcov(RegresionX)


Call:
lm(formula = y ~ x1 + x2)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.6671 -0.7791  0.0921  0.9816  4.0126 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)   
(Intercept)  15.7979     6.5574   2.409  0.02761 * 
x1            0.4399     0.1328   3.312  0.00412 **
x2            0.2398     0.1928   1.244  0.23044   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.865 on 17 degrees of freedom
Multiple R-squared:  0.5309,	Adjusted R-squared:  0.4757 
F-statistic: 9.621 on 2 and 17 DF,  p-value: 0.001606


(Intercept) x1          x2         
(Intercept) 42.9992368  -0.10046810 -1.07995395
x1          -0.1004681   0.01764284 -0.01062163
x2          -1.0799540  -0.01062163  0.03717570

Ahora se quiere cambiar las variables de la siguiente manera: 

$$
    W_1:\text{cintura más cadera}
$$
$$
    W_1=X_1+X_2
$$

$$
    W_2:\text{cintura menos cadera}
$$
$$
    W_2=X_1-X_2
$$


In [2]:
#Regresión lineal con W1 y W2
w1 <- x1+x2
w2 <- x1-x2

RegresionW <- lm(y ~ w1 + w2)
summary(RegresionW)
vcov(RegresionW)


Call:
lm(formula = y ~ w1 + w2)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.6671 -0.7791  0.0921  0.9816  4.0126 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)   
(Intercept) 15.79785    6.55738   2.409  0.02761 * 
w1           0.33985    0.09162   3.709  0.00174 **
w2           0.10003    0.13790   0.725  0.47808   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.865 on 17 degrees of freedom
Multiple R-squared:  0.5309,	Adjusted R-squared:  0.4757 
F-statistic: 9.621 on 2 and 17 DF,  p-value: 0.001606


(Intercept) w1           w2          
(Intercept) 42.9992368  -0.590211024  0.489742926
w1          -0.5902110   0.008393819 -0.004883215
w2           0.4897429  -0.004883215  0.019015446

##### a. Se quiere encontrar los vectores $\hat{Y}$ con los modelos que tiene las variables $X$ y las variables $W$

Por un lado, se tiene el modelo de las variables $X$ de la siguiente manera 
$$
Y=X \beta + \varepsilon
$$

$$
\hat{Y}=X\hat{\beta}
$$

$$
\hat{Y}=X\left(X^tX\right)^{-1}X^tY
$$

Por otro lado, se tiene el modelo, pero con las variables $W$: 
$$
Y=W \theta + \varepsilon
$$

$$
\hat{Y}=W\hat{\theta}
$$

$$
\hat{Y}=W\left(W^tW\right)^{-1}W^tY
$$


In [3]:
#Y gorro con X1 y X2
X <- cbind(1, x1, x2)
xtx <- t(X) %*% X
H <- solve(xtx)
P <- X %*% H %*% t(X)
beta_hat <- H %*% t(X) %*% y
y_hat <- P %*% y
sigma2 <- sum((y - y_hat) ^ 2) / (20 - 3)

print(y_hat)

          [,1]
 [1,] 36.66866
 [2,] 38.74754
 [3,] 37.62795
 [4,] 39.66707
 [5,] 36.46860
 [6,] 38.98736
 [7,] 40.38654
 [8,] 38.58724
 [9,] 37.86777
[10,] 39.22718
[11,] 34.74883
[12,] 35.66836
[13,] 35.22848
[14,] 33.62925
[15,] 36.34807
[16,] 36.10824
[17,] 35.90818
[18,] 36.10824
[19,] 37.70748
[20,] 34.30895


In [4]:
#Y gorro con W1 y W2
W <- cbind(1, w1, w2)
wtw <- t(W) %*% W
Hw <- solve(wtw)
Pw <- W %*% Hw %*% t(W)
beta_hatw <- Hw %*% t(W) %*% y
y_hatw <- Pw %*% y
sigma2w <- sum((y - y_hatw) ^ 2) / (20 - 3)

print(y_hatw)

          [,1]
 [1,] 36.66866
 [2,] 38.74754
 [3,] 37.62795
 [4,] 39.66707
 [5,] 36.46860
 [6,] 38.98736
 [7,] 40.38654
 [8,] 38.58724
 [9,] 37.86777
[10,] 39.22718
[11,] 34.74883
[12,] 35.66836
[13,] 35.22848
[14,] 33.62925
[15,] 36.34807
[16,] 36.10824
[17,] 35.90818
[18,] 36.10824
[19,] 37.70748
[20,] 34.30895


Como se puede observar que el vector $\hat Y$ es exactamente el mismo entre los dos modelos, para el modelo con las variables X y para el modelo con las variables W, lo cual muestra que efectivamente, como se mencionó en el literal anterior, los $\hat Y$ deben ser y son iguales incluso cuando las columnas del vector $X$ y del vector $W$ no son iguales pero sí son independientes y es por esto tambien que se construye el mismo subespacio de proyección tal como tambien se comprobó anteriormente. 

##### b. Contrastar un intervalo de confianza para el valor esperado del contorno del pecho

Para el intervalo de confianza se utiliza lo siguiente: 
$$
IC:\left[\hat{Y}-t_{\alpha/2,gle}*\sqrt{{\hat{\sigma}}^2C^T\left(X^TX\right)^{-1}C}\right]
$$

Para el caso de X: 
$$
c=\left(\begin{matrix}1&30&30\\\end{matrix}\right)
$$

$$
{I.C}_x=\left[{\hat{\beta}}_0+30{\hat{\beta}}_1+30{\hat{\beta}}_2-\ t_{\frac{\alpha}{2},gle}\sqrt{{\hat{\sigma}}^2c^T\left(X^TX\right)^{-1}c};\ {\hat{\beta}}_0+30{\hat{\beta}}_1+30{\hat{\beta}}_2+t_{\frac{\alpha}{2},gle}\sqrt{{\hat{\sigma}}^2c^T\left(X^TX\right)^{-1}c}\right]
$$


Con el vector $\beta$ hallado en el literal anterior, $gle = 17$ y $\alpha = 0.05$.

In [5]:
#Intervalo de confianza con X1 y X2
cx <- c(1,30,30)
limitex_inf <- t(cx) %*% beta_hat - qt(0.975,17) * sqrt(sigma2 * (t(cx) %*% H %*% cx))
limitex_sup <- t(cx) %*% beta_hat + qt(0.975,17) * sqrt(sigma2 * (t(cx) %*% H %*% cx))

IC1 <- c(limitex_inf,limitex_sup)
IC1

[1] 32.92618 39.45184

$$
    IC_x:\left[32.926; 39.452\right]
$$

Y Para $W$ es:
$$
c=\left(\begin{matrix}1&60&0\\\end{matrix}\right)
$$

$$
{I.C}_w=\left[{\hat{\theta}}_0+60{\hat{\theta}}_1+0{\hat{\theta}}_2-\ t_{\frac{\alpha}{2},gle}\sqrt{{\hat{\sigma}}^2p^T\left(X^TX\right)^{-1}p};\ {\hat{\theta}}_0+60{\hat{\theta}}_1+0{\hat{\theta}}_2+t_{\frac{\alpha}{2},gle}\sqrt{{\hat{\sigma}}^2p^T\left(X^TX\right)^{-1}p}\right]
$$



In [6]:
#IC con W1 y W2
cw <- c(1,60,0)
limitew_inf <- t(cw)%*%beta_hatw - qt(0.975,17) * sqrt(sigma2w * (t(cw) %*% Hw %*% cw))
limitew_sup <- t(cw)%*%beta_hatw + qt(0.975,17) * sqrt(sigma2w * (t(cw) %*% Hw %*% cw))

IC2 <- c(limitew_inf, limitew_sup)
IC2

[1] 32.92618 39.45184

$$
    IC_{w}:\left[32.926; 39.452\right]
$$

Estos intervalos de confianza nos muestra que con un nivel de confianza del 95% el valor esperado de $Y$, es decir el valor esperado de la medida del pecho va a estar entre 32.926 y 39.452 tanto para el caso en el que la `cintura` $= 30$ y la `cadera` $= 30$ como en el caso en el que el tamaño es $60$ y la forma es $0$. 

Esto quiere decir que la proyección del subespacio es la misma, por lo que el valor esperado de cada valor de $\hat Y$ es el mismo, sin importar la transformación que se le haga al vector $X$

### Problema 6. Fundamentos de Álgebra para Modelos Lineales.

Sean $a$, $z$ en $\mathbb{R}^p$ y $M$ una matriz en $\mathbb{R}^{p \times p}$. Se definen las funciones:

$$
\alpha_1 = a^Tz \quad \alpha_2 = z^T\mathbb{M}z
$$

Demuestre:

#### I. $\frac{d\alpha_1}{dz} = a$

Para expresar la derivada de $\alpha_1$ respecto al vector $z$ es necesario hallar la expresión resultante del producto. Al operar los dos vectores se tendría lo siguiente:

$$
    a^Tz =
        \begin{bmatrix}
            a_1 & a_2 & \cdots & a_p
        \end{bmatrix}
        \begin{bmatrix}
            z_1 \\
            z_2 \\
            \vdots \\
            z_p
        \end{bmatrix}
        = a_1 z_1 + a_2 z_2+ \cdots + a_p z_p.
$$

Al obtener la expresión esclar, es posible utilizar la definición del Jacobiano para hallar el vector de derivadas

$$
    \frac{d \alpha_1}{dz} =
        \begin{bmatrix}
            \frac{d \alpha_1}{dz_1} \\ 
            \frac{d \alpha_1}{dz_2} \\
            \vdots \\
            \frac{d \alpha_1}{dz_p}
        \end{bmatrix}.
$$

Es decir,

$$
    \frac{d \alpha_1}{dz} =
        \begin{bmatrix}
        \frac{d}{dz_1} (a_1z_1 + a_2z_2+ \cdots + a_pz_p) \\
        \frac{d}{dz_2} (a_1z_1 + a_2z_2+ \cdots + a_pz_p) \\
        \vdots \\
        \frac{d}{dz_p} (a_1z_1 + a_2z_2+ \cdots + a_pz_p)
    \end{bmatrix}.
$$

En cada componente $i$ del vector, la derivada será igual a $a_i$, dado que los demás términos serán constantes. Es decir, se obtendrá el siguiente vector.

$$
    \frac{d \alpha_1}{dz} =
        \begin{bmatrix}
            a_1 \\
            a_2 \\
            \vdots \\
            a_p
        \end{bmatrix}.
$$

De esta forma, es posible afirmar que 

$$
\frac{d \alpha_1}{dz} = a.
$$ 

#### II. $\frac{d\alpha_2}{dz} = (\mathbb{M} + \mathbb{M}^T)z$.

Nuevamente, para expresar la derivada de $\alpha_2$ será necesario hallar la expresión escalar resultante del producto matricial. Esto, lo descompondremos en dos pasos. En primer lugar, el producto $z^T\mathbb{M}$ se vería de la siguiente manera:

$$
    z^T\mathbb{M} =
        \begin{bmatrix}
            z_1 & z_2 & \cdots & z_p
        \end{bmatrix}
        \begin{bmatrix}
            m_{11} & m_{12} & \cdots & m_{1p} \\
            m_{21} & m_{22} & \cdots & m_{2p} \\
            \vdots & \vdots & \ddots & \vdots \\
            m_{p1} & m_{p2} & \cdots & m_{pp}
        \end{bmatrix}.
$$

Al observar la primera componenete del vector resultante, se vería la siguiente expresión:

$$
z_1m_{11} + z_2m_{21} + \cdots + z_pm_{p1}.
$$

Consecuentemente, la expresión del $i$-ésimo componente de este vector será

$$
z_1m_{1i} + z_2m_{2i} + \cdots + z_pm_{pi}.
$$

Esta expresión se puede compactar con una sumatoria. Es decir, el vector se puede ver como:

$$
    z^T\mathbb{M} =
    \begin{bmatrix}
        \sum_{i=1}^{p}{z_im_{i1}} & \sum_{i=1}^{p}{z_im_{i2}} & \cdots & \sum_{i=1}^{p}{z_im_{ip}}
    \end{bmatrix}.
$$

Para encontrar la expresión escalar, hace falta multiplicar el vector que se obtuvo por el vector $z$. Es decir,

$$
    z^T\mathbb{M}z =
    \begin{bmatrix}
        \sum_{i=1}^{p}{z_im_{i1}} & \sum_{i=1}^{p}{z_im_{i2}} & \cdots & \sum_{i=1}^{p}{z_im_{ip}}
    \end{bmatrix}
    \begin{bmatrix}
        z_1 \\
        z_2 \\
        \vdots \\
        z_p
    \end{bmatrix}.
$$

Así, la expresión resultante es:

$$
    z^T\mathbb{M}z = z_1\sum_{i=1}^{p}{z_im_{i1}} + z_2\sum_{i=1}^{p}{z_im_{i2}} + \cdots + z_p \sum_{i=1}^{p}{z_im_{ip}}.
$$

Una vez obtenemos la expresion escalar, empleamos el jacobiano para hallar la derivada de la función matricial. Para esto, veremos el primer componente del vector de derivadas, es decir $\frac{d(z^T\mathbb{M}z)}{dz_1}$. Para hallar la derivada, eliminaremos en primer lugar todos terminos que no contengan $z_1$ ya que serán constantes. Así, los terminos que quedan (sin derivar aún) son:

$$
    z_1\sum_{i=1}^{p}{z_im_{i1}} + z_2(z_1m_{12}) + \cdots + z_p(z_1m_{1p}).
$$

La primera sumatoria se puede expandir para obtener lo siguiente:

$$
    z_1(z_1m_{11} + z_2m_{21} + \cdots + z_pm_{p1}) + z_2(z_1m_{12}) + \cdots + z_p(z_1m_{1p}).
$$

Al expandir la expresión se obtendría lo siguiente:

$$
(z_1z_1m_{11} + z_1z_2m_{21} + \cdots + z_1z_pm_{p1}) + z_2z_1m_{12} + \cdots + z_pz_1m_{1p}.
$$

En este momento, se puede ver que la derivada sería la siguiente expresión:

\begin{align*}
        \frac{d\alpha_2}{dz_1} &= (2z_1m_{11} + z_2m_{21} + \cdots + z_pm_{p1}) + z_2m_{12} + \cdots + z_pm_{1p} \\
        &= z_1(m_{11} + m_{11}) + z_2(m_{21} + m_{12}) + \cdots + z_p(m_{p1} + m_{1p}).
\end{align*}

Al generalizar, la $i$-ésima derivada sería

$$
    \frac{d\alpha_2}{dz_i} = z_1(m_{1i} + m_{i1}) + z_2(m_{2i} + m_{i2}) + \cdots + z_p(m_{pi} + m_{ip}).
$$

Al organizar el vector de derivadas, es posible comprobar que es igual al producto $(\mathbb{M} + \mathbb{M}^T)z$

$$
    \frac{d\alpha_2}{dz} =
        \begin{bmatrix}
            z_1(m_{11} + m_{11}) + z_2(m_{21} + m_{12}) + \cdots + z_p(m_{p1} + m_{1p}) \\
            z_1(m_{12} + m_{i2}) + z_2(m_{22} + m_{22}) + \cdots + z_p(m_{p2} + m_{2p}) \\
            \vdots \\
            z_1(m_{1p} + m_{p1}) + z_2(m_{2p} + m_{p2}) + \cdots + z_p(m_{pp} + m_{pp})
        \end{bmatrix}.
$$

### Problema 7. Fundamentos de Inferencia Paramétrica.

Se tiene una muestra aleatoria $Y_1,Y_2,\cdots, Y_n$ de una población $Y\sim \text{Exponencial}(\theta)$ con función de densidad $f(x)=\frac{1}{\theta}\cdot e^\frac{-x}{\theta}$, donde $E\left[Y\right]=\theta $ y $Var\left(Y\right)=\theta^2$.

#### I. Halle el estimador de máxima verosimilitud $\hat{\theta}$

Para hallar el estimador de máxima verosimilitud se deben seguir los siguientes pasos: 

1.	Función de verosimilitud
$$
L\left(\theta;Y_1,\ Y_2,\ \cdots,\ Y_n\right)=\prod_{i=1}^{n}f_x\left(Y_i;\theta\right)=\prod_{i=1}^{n}\frac{1}{\theta}\cdot e^\frac{-{Y}_i}{\theta}=\left(\frac{1}{\theta}\right)^n\cdot\ e^{- \frac{1}{\theta}\cdot\sum_{i=1}^{n}Y_i}
$$

2.	Logaritmo natural de la función de verosimilitud 
$$
\ln{\left(L\left(\theta;Y_1,\ Y_2,\ \cdots,\ Y_n\right)\right)}=\ln{\left[\left(\frac{1}{\theta}\right)^n\cdot e^{\frac{1}{\theta}\cdot\sum_{i=1}^{n}Y_i}\right]}
=\ln{\left(\left(\frac{1}{\theta}\right)^n\right)}+\ln{\left(e^{-\frac{1}{\theta}\cdot\sum_{i=1}^{n}Y_i}\right)}
=-n\cdot\ln{\left(\theta\right)}-\frac{1}{\theta}\cdot\sum_{i=1}^{n}Y_i
$$

3.  Derivar respecto al parámetro
$$
\frac{\partial}{\partial\theta}\left[-n\cdot\ln{\left(\theta\right)}-\frac{1}{\theta}\cdot\sum_{i=1}^{n}Y_i\right]
=\frac{\partial}{\partial\theta}\left[-n\cdot\ln{\left(\theta\right)}\right]-\frac{\partial}{\partial\theta}\left[\frac{1}{\theta}\cdot\sum_{i=1}^{n}Y_i\right]
=-\frac{n}{\theta}+\frac{\sum_{i=1}^{n}Y_i}{\theta^2}
$$

4. Igualar a 0 y despejar el parámetro
\begin{align*}
    -\frac{n}{\theta}+\frac{\sum_{i=1}^{n}Y_i}{\theta^2} &= 0\\
    \frac{\sum_{i=1}^{n}Y_i}{\theta^2} &=\frac{n}{\theta} \\
    \frac{\sum_{i=1}^{n}Y_i}{\theta}&=n \\
    \frac{\sum_{i=1}^{n}Y_i}{n} &= \hat{\theta} \\
    \bar{Y} &= \hat{\theta}
\end{align*}

El procedimiento anterior mostró que el estimador de máxima verosimilitud para la función de densidad dada es $\hat{\theta}=\bar{Y}$.

#### II. 	Comente sobre la distribución de $\frac{\hat{\theta}-{\theta}}{\sqrt{\left(\frac{{\theta}^\mathbf{2}}{t{n}}\right)}\ }\ $cuando $n$ representa un tamaño de muestra grande. 

Para este caso en donde sucede que $n$ representa un número muy grande se puede hacer uso del teorema del límite central (TLC), el cual establece que si se tiene una población con un tamaño de muestra mayor a 30 (es decir $n>30$) entonces la distribución de la media asemeja a una distribución normal cuya media es $\mu$ y la varianza es $\frac{\sigma^2}{n}$. Para el caso del TLC se pueden tener dos consideraciones, la primera cuando lo que se quiere es un promedio y la segunda cuando se quiere una suma.

Particularmente para $Y\sim \text{Exponencial(\theta)}$, $\bar{Y} \sim \text{Normal(\)}$

Estas se expresan de la siguiente manera: 

$$
\lim_{n\rightarrow\infty}{\left(\bar{x}\right) \sim \text{Normal}\left(\mu,\frac{\sigma^{2}}{n}\right)}
$$

$$
\lim_{n\rightarrow\infty}{\left(\frac{\bar{{x}}-{\mu}}{\frac{{\sigma}}{\sqrt{n}}}\right) \sim \text{Normal}\left(0,1\right)}
$$

$$
\lim_{n\rightarrow\infty}{\left(\frac{\bar{x}-n\mu}{\frac{\sigma^2}{\sqrt n}}\right) \sim Normal\left(0,1\right)}
$$

Así para este caso en particular de la distribución exponencial, sí n es un número muy grande entonces se puede estandarizar como una distribución normal de la siguiente manera:
$$
{{lim}}{{n}\rightarrow\infty}{\left(\frac{\hat{{\theta}}-{\theta}}{\frac{{\theta}}{\sqrt{n}}}\right)\ \sim \ {Normal}\ (\mathbf{0},\mathbf{1})}
$$

Es decir que cuando $n>>$ entonces…
$$
{z}=\frac{\hat{{\theta}}-{\theta}}{\frac{\hat{{\theta}}}{\sqrt{n}}}\ \sim \ {Normal}\ (\mathbf{0},\mathbf{1})
$$

#### III. Consulte sobre el teorema de Slutsky y úselo junto con la información del numeral anterior para construir un intervalo de confianza $(1-\alpha)$ para el parámetro $\theta$. 

El teorema de Slutsky plantea algunas propiedades para tener en cuenta en las operaciones de variables aleatorias sobre sucesiones convergentes, es decir que sus límites tienden a un número finito perteneciente a los números reales. 
Sí se tiene una sucesión de variables aleatorias $X_n$ y $Y_n$, en donde $X_n$ debe ser convergente en distribución y $Y_n$ convergente en probabilidad/Distribución a una constante c.

Propiedad 1: $X_n+Y_n$  $\underrightarrow{d}$  $X+c$

Propiedad 2: $X_n\ast\ Y_n$  $\underrightarrow{d}$  $cX$

Propiedad 3: $:\frac{X_n}{Y_n}$   $\underrightarrow{d}$  $\frac{X}{c}$



                                                        … en donde d corresponde a la convergencia en distribución



Para construir el intervalo de confianza se tiene en cuenta el resultado del punto anterior y las propiedades mencionadas para poder afirmar que en efecto la distribución que se tenía en el punto anterior converge a una normal estándar con parámetros (media=1 y varianza=0)
$$
{z}=\frac{\hat{{\theta}}-{\theta}}{\frac{\hat{{\theta}}}{\sqrt{n}}}\ \underrightarrow{d}\ {Normal}\ (\mathbf{0},\mathbf{1})
$$

Ya para la construcción del Intervalo de Confianza (IC) se tiene en cuenta las propiedades de la distribución normal que establecen que debe existir un valor  $z_\frac{\alpha}{2}$ tal que: 
$$
P\left(-z_\frac{\alpha}{2}\le Z\le z_\frac{\alpha}{2}\right)=1-\alpha
$$


Sí se remplazan los valores se tiene que: 
$$
P\left(-z_\frac{\alpha}{2}\le\frac{\hat{{\theta}}-{\theta}}{\frac{\hat{{\theta}}}{\sqrt\{n}}\le z_\frac{\alpha}{2}\right)=1-\alpha
$$

Sí se multiplica la desigualdad por $\frac{\sigma}{\sqrt n}$ queda… 
$$
P\left(-z_\frac{\alpha}{2}\frac{\hat{{\theta}}}{\sqrt{n}}\le\hat{{\theta}}-{\theta}\le z_\frac{\alpha}{2}\frac{\hat{{\theta}}}{\sqrt{n}}\right)=1-\alpha
$$


Luego se pasa a restar el estimador $\hat{\theta}$
$$
P\left(-z_\frac{\alpha}{2}\frac{\hat{{\theta}}}{\sqrt{n}}-\hat{{\theta}}\le\left(-{\theta}\right)\le z_\frac{\alpha}{2}\frac{\hat{{\theta}}}{\sqrt{n}}-\hat{{\theta}}\right)=1-\alpha
$$


Y ahora se se multiplica por (-1) y se reajusta el orden 
$$
P\left(\hat{{\theta}}+z_\frac{\alpha}{2}\frac{\hat{{\theta}}}{\sqrt{n}}\le{\theta}\le\ \hat{{\theta}}{-z}_\frac{\alpha}{2}\frac{\hat{{\theta}}}{\sqrt{n}}\right)=1-\alpha
$$


Por lo que el intervalo de confianza queda como se presenta a continuación:
$$
IC:\left[\hat{{\theta}}+z_\frac{\alpha}{2}\frac{\hat{{\theta}}}{\sqrt{n}};\hat{{\theta}}{-z}_\frac{\alpha}{2}\frac{\hat{{\theta}}}{\sqrt{n}}\right]
$$

#### IV. 	Suponiendo que el número de entradas que ocurren al baño del 5 piso del edificio ML, se comportan como un proceso Poisson con tasa $\frac{\mathbf{1}}{\mathbf{\theta}}$. Y suponiendo que, en una semana, el promedio de entradas por hora es de 21.4 y que el día laboral tiene 12 horas.  Se construye un intervalo de confianza para la media del tiempo entre arribos con una confianza del 95%

Sí se tiene un evento que ocurre al azar varias veces a lo largo del tiempo y de manera continua, y se hace el supuesto de que los tiempos de ocurrencia son independientes entre sí y todos tienen una distribución exponencial de parámetro $\frac{1}{\theta}$. 
Se tiene entonces que:
$$
\hat{{\theta}}=\frac{1}{\theta}=\frac{1}{21.4}=0.0467
$$

$$
n=\frac{12 horas}{día} * \frac{5días}{semana}= \frac{60horas}{semana}
$$

$$
\frac{\alpha}{2}=\frac{0.05}{2}=0.025
$$

Con lo anterior se reemplazan los valores del intervalo de confianza del literal anterior y se obtiene que: 
$$
IC:\left[\mathbf{0}.\mathbf{0467}-1.96)\frac{\mathbf{0}.\mathbf{0467}}{\sqrt{\mathbf{60}}};\ \mathbf{0}.\mathbf{0467}+1.96)\frac{\mathbf{0}.\mathbf{0467}}{\sqrt{\mathbf{60}}}\right]
$$

$$
IC:\left[0.0349;0.0586\right]
$$

El resultado anterior quiere decir que con una confianza del 95%, el parámetro teta se encuentra entre 0.0349 y 0.0586, es decir que la media del tiempo en horas entre los arribos al baño del 5 piso del edificio ML está entre 0.0349 y 0.0586.


### Problema 8. Proyecciones y Distribuciones.

Teniendo en cuenta que $\mathbb{Y}\sim \text{Normal}_{40}(\mathbb{X}\beta,\sigma^2\mathbb{I}_{40x40})$ y que la matrix de proyección $\mathbb{P_{\Omega}}=\mathbb{X}(\mathbb{X}^t\mathbb{X})^{-1}\mathbb{X}^t$.

1. Demostrar porqué $(\mathbb{Y} - \mathbb{X}\beta)^t(\mathbb{I} - \mathbb{P_{\Omega}})(\mathbb{Y} - \mathbb{X}\beta)=||\mathbb{Y} - \mathbb{P_{\Omega}}\mathbb{Y}||^2 = || \mathbb{Y} - \hat{\mathbb{Y}}||^2$.

Para probar que las tres expresiones son las mismas, proponemos partir de la primera. Esta expresa la norma al cuadrado del vector entre las observaciones $\mathbb{Y}$ y el estimador $\hat{\mathbb{Y}}$. Sin embargo, el estimador que se tiene de las observaciones se puede entender como la proyección del vector $\mathbb{Y}$ sobre el sub-espacio $\Omega \subset \mathbb{R}^p$ generado por la combinación lineal de las columnas de la matrix $\mathbb{X}$. Es decir, empleando la defición de la matrix de proyección $P_{\Omega}$, se puede reescribir $\hat{\mathbb{Y}}$ como $\mathbb{P_{\Omega}}\mathbb{Y}$. Así, derivamos la primera igualdad. 

$$
||\mathbb{Y} - \mathbb{P_{\Omega}}\mathbb{Y}||^2 = || \mathbb{Y} - \hat{\mathbb{Y}}||^2
$$

En cuanto a la tercera expresión, se puede operar para comprobar que se llegará al mismo resultado. Para ver esto, proponemos operar el producto de las dos matrices de la derecha. Es decir, 

$$(\mathbb{I} - \mathbb{P_{\Omega}})(\mathbb{Y} - \mathbb{X}\beta) = \mathbb{I}\mathbb{Y} - \mathbb{I}\mathbb{X}\beta - \mathbb{P}_{\Omega}\mathbb{Y} + \mathbb{P_{\Omega}}\mathbb{X}\beta$$

En primer lugar, los vectores multiplicados por la identidad serán iguales. Por otra parte, se reemplazará el termino $\mathbb{X}\beta$ por $\hat{\mathbb{Y}}$. Así, se tiene lo siguiente.

$$
 \mathbb{Y} - \hat{\mathbb{Y}} - \mathbb{P}_{\Omega}\mathbb{Y} + \mathbb{P_{\Omega}}\hat{\mathbb{Y}}
 $$

Para continuar, son necesarios dos conceptos asociados a la matriz de proyección $\mathbb{P_{\Omega}}$. El primero, es que por definición $\mathbb{P}_{\Omega}\mathbb{Y} = \hat{\mathbb{Y}}$ pues dicha matriz es aquella que proyecta los vectores $\mathbb{Y}$ en el sub-espacio $\Omega$. El segundo, que sigue el mismo proncipio, es que la proyección de un vector sobre el sub-espacio al que pertenece es él mismo. De esta manera, $\mathbb{P_{\Omega}}\hat{\mathbb{Y}} = \hat{\mathbb{Y}}$. Al reemplazar las expresiones presentadas previamente, se obtiene la siguiente expresión. 

$$ 
\mathbb{Y} - \hat{\mathbb{Y}} - \hat{\mathbb{Y}} + \hat{\mathbb{Y}} = \mathbb{Y} -  \hat{\mathbb{Y}}
$$

Evaluando el término hallado en la expresión original, se tendría que 
 
$$
(\mathbb{Y} - \mathbb{X}\beta)^t(\mathbb{I} - \mathbb{P_{\Omega}})(\mathbb{Y} - \mathbb{X}\beta)=(\mathbb{Y} - \mathbb{X}\beta)^t(\mathbb{Y} -  \hat{\mathbb{Y}})
$$

Utilizando la misma definición que se ha empleado para toda la explicación, se puede reemplazar $\mathbb{X}\beta$ por $\hat{\mathbb{Y}}$ y esta es la operación matricial para hallar la norma de un vector al cuadrado $||\mathbb{G}||^2=\mathbb{G}^t\mathbb{G}$


2. Encontrar la distribución de $\frac{1}{\sigma^2}||\mathbb{Y} - \hat{\mathbb{Y}}||^2$.

Del literal anterior, se puede establecer que $||\mathbb{Y} - \hat{\mathbb{Y}}||^2 = (\mathbb{Y} - \mathbb{X}\beta)^t(\mathbb{Y} - \mathbb{X}\beta)$. A partir de esto, vamos a basarnos en la siguiente definición para hallar la distribución. 

-   Si el vector aleatorio $\mathbb{W} \sim Normal_p(\mu,\Sigma)$, entonces la variable definida como $(\mathbb{W} - \mu)^T\Sigma^{-1}(\mathbb{W} - \mu)$ se distribuye $x^2(p)$.


Empleando esta definición, es posible traducir la expresión anterior para que cumpla con la condición. Es decir, $\mathbb{W} = \mathbb{Y}$, $\mu = \mathbb{X}\beta$ y $\Sigma = \sigma^2 \mathbb{I}$. Así, se obtiene lo siguiente:

$$
(\mathbb{Y} - \mathbb{X}\beta)^T(\sigma^2 \mathbb{I})^{-1}(\mathbb{Y} - \mathbb{X}\beta)
$$

Operamos el inverso para los dos terminos del producto de la mitad para obtener lo siguiente. Adicionalmente, utilizamos la definición de la proyección para reemplazar $\mathbb{X}\beta$ por $\hat{\mathbb{Y}}$

$$
(\mathbb{Y} - \hat{\mathbb{Y}})^T(\sigma^2)^{-1} (\mathbb{I})^{-1}(\mathbb{Y} - \hat{\mathbb{Y}}) = (\mathbb{Y} - \hat{\mathbb{Y}})^T \frac{\mathbb{I}}{\sigma^2} (\mathbb{Y} - \hat{\mathbb{Y}})
$$

Finalmente, se reorganiza la expresión para ver que el producto entre los dos paréntesis es la norma al cuadrado del vector $\mathbb{Y} - \hat{\mathbb{Y}}$. Es decir, que


$$
(\mathbb{Y} - \hat{\mathbb{Y}})^T \frac{\mathbb{I}}{\sigma^2} (\mathbb{Y} - \hat{\mathbb{Y}}) = \frac{1}{\sigma^2}||\mathbb{Y} -  \hat{\mathbb{Y}}||^2
$$

Por la definición que se dio al inicio de la explicación, esta expresión se distribuye Chi-cuadrado con p grados de libertad $x^2(p)$

3. Porqué si el modelo es significagivo no se distribuye como $x^2$ con 39 grados de libertad.

Ya se ve que la expresión $\frac{1}{\sigma^2}||\mathbb{Y} -  \hat{\mathbb{Y}}||^2$ se distribuye Chi-cuadrado con $p$ gradosd de libertad. En este caso, $p$ que es el número de parámetros a estimar $(k + 1)$ son 10. Por lo tanto, la distribución de $\frac{1}{\sigma^2}||\mathbb{Y} -  \hat{\mathbb{Y}}||^2$ sería una Chi-cuadrado con 10 grados de libertad, no 39.

## Parte B. Problemas Aplicados con Datos Reales.

### Problema 9. _Carseats Sales Data_.

Para construir el modelo de regresión lineal múltiple es necesario procesar los datos que se van a analizar. Lo primero será remover las variables categoricas para trabajar únicamente con las continuas. 

In [ ]:
# Uploading library and processing data
library(ISLR)
categorical_variables = c(7,10,11)
data=Carseats[,-categorical_variables] #remove non-continuous variables # nolint
head(data)
help(Carseats)

1. Estimar un modelo de regresión lineal e interpretar cada uno de los parámetros de la recta.

In [ ]:
model = lm(Sales ~ ., data = data)
print(model)

De acuerdo con los resultados obtenidos, se puede hacer el siguiente análisis sobre los coeficientes del modelo.

1. CompPrice: Por cada dolar que aumente el precio que cobran los proveedores, se estima que el valor esperado de las ventas aumentará en 0.0939 mil unidades.


2. Income: Por cada mil dólares que aumente el ingreso de la comunidad, se estima que el valor esperado de las ventas aumentará en 0.0129 mil unidades.


3. Por cada mil dolares que se aumente en la inversión de la publicidad, se estima que el valor esperado de las ventas aumentará en 0.131 mil unidades.


4. Por cada mil unidades que aumente la población, se estima que el valor esperado de las ventas disminuirá en 0.00012 mil unidades.


5. Por cada dolar que aumente el precio de las sillas, se estima que el valor esperado de las ventas disminuya en 0.0926 mil unidades.


6. Por cada año que aumente la población, se estima que el valor esperado de las ventas disminuirá en 0.044 mil unidades.


7. Por cada nivel que aumente la educación de la población, se estima que el valor esperado de las ventas disminuyan en 0.0399 mil unidades.

In [ ]:


model = lm(Sales~CompPrice + Income + Advertising + Price + Age, data = data) # nolint
summary(model)

### Problema 10. Problema Libre.

Se debe escoger unos datos que estén relacionados a un problema inferencial de su preferencia. Debe haber una variable de interés $(Y)$ y al menos 3 variables explicativas $(k \le 3)$. Recuerde que, para un problema de regresión, Y debe ser continua.



#### I.	Haga una descripción de los datos y del problema general sobre el cual quiere inferir. Muestre algunas gráficas que resuman el comportamiento de los datos.

Para este problema se utilizará el paquete de R “ISLR”, el cual significa Data for an Introduction to Statistical Learning with Applications in R. Dentro de este paquete se pueden encontrar diferentes tipos de datos, pero en este caso utilizaremos los datos Auto, que hacen referencia a un total de 392 datos de diferentes tipos de automóviles con sus algunas características. A continuación, se muestran las variables que están dentro de este paquete:

|Variable|Descripción|
|--------|-----------|
|mpg| Millas por galón|
|Cylinders| Número de cilindros que tiene el carro (entre 4 a 8)|
|Displacement| Desplazamiento del motor en pulgadas|
|Horsepower|Caballos de fuerza del motor|
|Weight|Peso del carro en libras|
|Acceleratios|Tiempo de aceleración de 0 a 60 millos por hora por segundo|
|Year|Año del modelo|
|Origin| Un factor que termina el origen del carro (1. Americano, 2. Europeo, 3. Japonés|
|Name| Nombre del auto|

Lo que se quiere es comprender los factores que caracterizan la aceleración de un carro, para lo cual es necesario eliminar las variables categóricas como lo es el Origin y el Name. Por otro lado, se debe asegurar que la base de datos no tenga información faltante, por lo que en realidad la base tiene 408 datos, pero el programa ya ha hecho la depuración correspondiente por lo que nuestro $n = 392.$

Ahora bien, ya con los datos organizados y ajustados es importante saber sobre el comportamiento de los mismos, por lo que se hizo un resumen de cada una de las variables en donde aparece el mínimo, el 1er cuantil, la mediana, la media, el 3er cuantil y el valor máximo.  Estos valores se muestran a continuación: 


In [ ]:
library("ISLR")
datos1 = Auto
datos=datos1[,c(1,2,3,4,5,6,7)]

y = datos$acceleration
x1 = datos$mpg
x2 = datos$cylinders
x3 = datos$displacement
x4 = datos$horsepower
x5 = datos$weight
x6 = datos$year

#Resumen de la base de datos 
summary(datos)

También para poder tener una mayor claridad sobre el comportamiento de cada una de las variables, se procede entonces a realizar 3 tipos de gráficos diferentes: histogramas, cajas y bigotes y por último gráficos de dispersión.

Con los histogramas vemos como se distribuyen los datos usando barras o segmentos y en cada barra se muestra la frecuencia de las observaciones


In [ ]:
#Grafícas básicas
#install.packages("ggplot2")
library(ggplot2)

#Histogramas
hist(y,main = "Histograma Aceleración")
hist(x1,main = "Histograma mpg")
hist(x2,main = "Histograma Cilindros")
hist(x3,main = "Histograma Desplazamiento")
hist(x4,main = "Histograma Caballos de fuerza")
hist(x5,main = "Histograma Peso")
hist(x6,main = "Histograma Año")


Otro de los tipos de gráficas que pueden ser muy útiles son los diagramas de cajas y bigotes, pues estos nos permiten ver la distribución de una variable usando los cuantiles. 

In [ ]:
#Diagramas de cajas y bigotes
Cajas_y_Bigotes1=boxplot(y,x1, main="Cajas & Bigotes Aceleración Vs mpg")
Cajas_y_Bigotes2=boxplot(y,x2, main="Cajas&Bigotes Aceleración Vs Cilindros")
Cajas_y_Bigotes3=boxplot(y,x3, main="Cajas&Bigotes Aceleración Vs Desplazamiento")
Cajas_y_Bigotes4=boxplot(y,x4, main="Cajas&Bigotes Aceleración Vs Caballos de fuerza")
Cajas_y_Bigotes5=boxplot(y,x5, main="Cajas&Bigotes Aceleración Vs Peso")
Cajas_y_Bigotes6=boxplot(y,x6, main="Cajas&Bigotes Aceleración Vs Año")

El diagrama de dispersión es supremamente útil pues nos permite ver la relación entre dos variables, en este caso la variable de respuestas y las explicativas.

In [ ]:
#Diagramas de dispersión
Dispersion1 = plot(x = datos$mpg, y = datos$acceleration)
Dispersion2 = plot(x = datos$cylinders, y = datos$acceleration)
Dispersion3 = plot(x = datos$displacement, y = datos$acceleration)
Dispersion4 = plot(x = datos$horsepower, y = datos$acceleration)
Dispersion5 = plot(x = datos$weight, y = datos$acceleration)
Dispersion6 = plot(x = datos$year, y = datos$acceleration)

#### II.	Se realizan las siguientes pruebas

Antes de realizar cualquier prueba es importante definir el modelo de regresión lineal con base en la variable de interés y sus correspondientes variables explicativas que decidimos conservar. 

$$
y=Aceleración
$$

$$
x_1=mpg
$$

$$
x_2=número de cilindros (cylinders)
$$

$$
x_3=desplazamiento(displacement)
$$

$$
x_4=Caballos\ de\ fuerza\ (Horsepower)
$$

$$
x_5=peso\ (Weight)
$$

$$
x_6=Año (Year)
$$

Y dadas estas variables, el modelo de regresión línea se puede expresar como: 

$$
Y=\beta_0+\beta_1X_1+\beta_2X_2+\beta_3X_3+\beta_4X_4+\beta_5X_5+\beta_6X_6+\varepsilon
$$

En el software de R el comando utilizado para expresar la regresión es el siguiente: 

$$
regresion\ =\ lm\left(y~x1+x2+x3+x4+x5+x6\right)
$$


In [ ]:
regresion = lm(y~x1+x2+x3+x4+x5+x6)
regresion

##### a. Significancia individual de cada variable

Ahora sí con las variables y el modelo de regresión lineal definido ahora sí se puede proceder a evaluar la significancia individual de cada una de las variables, para esto se debe utilizar una prueba t-Student de la siguiente manera: 
$$
H_0:\beta_j=0 
$$

$$
H_1:\beta_j\neq0
$$

$$
EP=\frac{\ \widehat{{\beta}_{j}}}{\sqrt{\widehat{{Var}}(\widehat{{\beta}_{j}})}}~t_{gl_{error}}\ 
$$

Se rechaza la hipótesis nula si:
$$
\left(|t_{EP}|\right)>tcrítico(\alpha /2,n-k-1)
$$

En R, este proceso se puede hacer muy fácilmente con el comando “summary”, por lo que se realiza: summary(regresion)


In [ ]:
summary(regresion)

Con estos resultados que podemos observar, nos damos cuenta que de manera individual, las variables Desplazamiento, Caballos de fuerza y peso son significativas para explicar la aceleración de un carro, esto lo sabemos pues el p-value de estas 3 variables es menor a la significancia de 5%. 

Adicionalmente nos arroja el valor de $R^2$, lo cual nos indica que tan bien mi modelo se ajusto a los datos. En este caso el valor del coeficiente de determinación fue $R^2=0.6139=61.39 \%$, recordando entonces que sí $R^2=100\%$ quiere decir que el modelo se ajusta muy bien a los datos, por lo que las variables $x$ explican totalmente la variación de $Y$. 
En este caso en particular este coeficiente no es tan alto, pero sí nos muestra que de alguna manera existen algunas relaciones y variables que en efecto explican la aceleración

$$
R^2=\frac{SSR}{SST}=1-\frac{SSE}{SST}
$$

$$
R^2=\frac{\sum_{t=1}^{T}{({\hat{Y}}_t-\bar{Y})}^2}{\sum_{t=1}^{T}{(Y_t-\bar{Y})}^2}
$$

##### b. Significancia global del modelo

Para establecer la significancia global del modelo es necesario en este caso utilizar una F, para la cual es necesario calcular los SSR y los SSE y con esto encontrar el MSE de la siguiente manera:

Para esta prueba yo voy a evaluar la siguiente hipotesis:
$$
𝐻_0:=\beta_1 =\beta_2 =\beta_3 = \beta_4 =\beta_5 =\beta_6=0
$$

$$
𝐻_1:Algún \ \beta_i  \ne 0
$$


$$
SSR=\sum_{i=1}^{n}{({\hat{Y}}_i-\bar{Y})}^2
$$

$$
SSE=\sum_{i=1}^{n}{(Y_i-{\hat{Y}}_i)}^2
$$

$$
MSE=\frac{SSE}{n-k-1}
$$

$$
MSR=\frac{SSR}{k}
$$

$$
F=\frac{MSR}{MSE}
$$

Luego se procede a calcular la prueba F, o F critico:
$$
F_{critico}(1- \alpha,k,n-k-1)
$$

In [ ]:
SSR = sum(fitted(regresion)-mean(y))^2
MSR = SSR/(6)

SSE = sum(y-mean(y))^2
MSE = SSE/(392-6-1)

#Calcular el estadístico de Prueba
F= MSR/MSE

#Caclcular el F crítico
Fcritico = qf(0.95,6,(392-6-1))

SSR
MSR
SSE
MSE
F
Fcritico

Dado que el estadístico de prueba $F>F_{crítico}$ pues $68.7215>2.1221$ dada una significancia del 5%, entonces se puede concluir que se rechaza la hipotesis nula por lo que el modelo es globalmente significativo.

##### c. Significancia parcial de algunas variables

Para hacer la significacia parcial se debe realizar una prueba F parcial en la cual voy a tener un modelo reducido, por lo que eliminaré las variables $x_1,\ x_2 \ y\  x_6$ dejando entonces el siguiente modelo reducido: 
$$
Y=\beta_0+\beta_3X_3+\beta_4X_4+\beta_5X_5+\varepsilon
$$

Es decir que debo tener la siguiente hipótesis: 
$$
𝐻_0:=\beta_1 =\beta_2 =\beta_6=0
$$

$$
𝐻_1:\ \beta_i  \ne 0 \ para \ algún\ i \in [1, 2, 6]
$$

y con esto se tiene el siguietne estadístico de prueba que hace referencia a la F parcial mencionada 

$$
F_{parcial}=\frac{\frac{SSR_{full}-SSR_{red}}{gl_{full}-gl_{red}}}{MSE_{full}}
$$
Luego se obtuvo el estadístico F crítico:
$$
F_{critico}(1- \alpha,k,n-k-1)
$$

Algo importante es que el modelo reducido no contiene esas 3 variables pues son las variables que no sos significativas para explicar la aceleración de un carro.

In [ ]:
# Se quita X1, X2 y X6 pues no es significativa
regresion_reducida = lm(y~x3+x4+x5)

SSR2=sum((fitted(regresion)-mean(y))^2)
SSE2=sum((fitted(regresion_reducida)-mean(y))^2)

#Calculo de la F (Estadistico de prueba)
F_parcial = (SSR2-SSE2)/3/MSE

#Calculo F critico
F_critico = qf(0.95,3,(392-3-1)) 

SSR2
SSE2
F_parcial
F_critico


Dado que el estadístico de prueba $F_{parcial}>F_{crítico}$ pues $2.877>2.627$ con un 95% de confianza se puede concluir que se rechaza la hipotesis nula por lo que las variables que excluimos en realidad son significativas para el modelo, por lo cual no es adecuado utilizar el modelo reducido 

##### d. Una combinación lineal de betas (que tenga sentido dentro del contexto del problema)

Para este punto queremos saber sí el efecto del número de cilindros es igual al efecto de los caballos de fuerza en la aceleración de un carro. 
Para poder realizar esto debemos considerar la siguiente hipotesis: 

$$
{H_o:\ \beta}_2-\beta_4=0
$$

$$
H_1:\beta_2-\beta_4\neq0
$$

Y para esta pruea se requiere un estadístico de prueba t como se muestra a continuación: 
$$
t=\frac{\widehat{\beta_2}-\widehat{\beta_4}\ -a }{\sqrt{MSE\cdot\left(\widehat{var}\left(\widehat{\beta_2}\right)+\widehat{var}\left(\widehat{\beta_4}\right)\right)}}
$$

$$
t=\left(\sum{C_j\ast\widehat{{\beta}_j}}-a\right)/\left(\sqrt{\widehat{var}\left(\sum C_j\right)\widehat{{\beta}_j}}\right)
$$
La t crítica de la prueba corresponde a:

$$
t_{1-\frac{\alpha}{2},\ n-k-1}
$$

In [ ]:
Cj =c(0,0,1,0,-1,0,0)
a=0

x=cbind(rep(1,392),x1,x2,x3,x4,x5,x6)
xtx=t(x)%*%x
H=solve(xtx)
beta_hat=H%*%t(x)%*%y

parte1=t(Cj)%*%beta_hat-a
parte2=sqrt(MSE*(t(Cj)%*%H%*%Cj))

EPt=parte1/parte2
EPt

t_critica=qt(0.975,392-6-1)
t_critica

Dado que el estadístico de prueba $t<t_{crítica}$ pues $-6.14^{13}<1.966$ con un 95% de confianza se puede concluir que no se rechaza la hipotesis nula, por lo que el efecto del número de cilindros es igual al efecto de los caballos de fuerza en la aceleración de un carro. 